sql/atxs/atxs.go

```diff
@@ -926,3 +926,44 @@ func SetUnits(db sql.Executor, atxID types.ATXID, id types.NodeID, units uint32)
 	)
 	return err
 }
+
+// AtxWithPrevious returns the ATX ID that has the given ATX ID as its previous ATX.
+func AtxWithPrevious(db sql.Executor, prev, not types.ATXID, id types.NodeID) (types.ATXID, error) {
+	var (
+		atxid types.ATXID
+		rows  int
+		err   error
+	)
+	if prev == types.EmptyATXID {
+		rows, err = db.Exec("SELECT id FROM atxs WHERE pubkey = ?1 AND prev_id IS NULL AND id != ?2;",
```

Comment:
`<>` is preferred to `!=` in SQL b/c it's a part of the ANSI standard

Reply:
Also, I'm not fully sure this inequality check is better done in SQL than in the Go code, it's a bit hard to say whether it will not confuse the query optimizer or not. It would be nice to verify the performance of this query on an up-to-date DB, and maybe also do `id != ...` check in the code

Reply:
We only need to sort the ATXs that have the same pubkey and prev_id by timestamp and pick the oldest, also no need to distinguish by `prev` and `not` is also not needed - since it will always be newer than another ATX that is already in the DB.

Reply:
Updated my previous comment to make clearer what I mean with a different strategy and query.

Reply:
I simplified the query not to filter out `not`. I realized it was unnecessary and explained this in a comment in the ATX handler. If there are more than 1 ATXs with the same `prev_id` in the DB, we don't need to generate the proof as it was already generated in the past. Hence, a simple check comparing the IDs of the processed ATX and the retrieved one is sufficient.